src/lib/components/TableOfContents/index.js

```diff
@@ -88,8 +91,9 @@ class TableOfContents extends BaseStateElement {
     this.observer.disconnect();
   }
 
-  onStateChanged({isTocOpened}) {
+  onStateChanged({isTocOpened, currentLanguage}) {
     this.opened = isTocOpened;
+    this.locale = currentLanguage;
```

Comment:
This is fine for now but since we are no longer a SPA, and also language changes always caused a full reload, this could just be static.